### ISSUE TITLE
ML.Dev.Pred.Category.Sig function error

### ISSUE THREAD

**Issue by kerwin223** [l-magnificence/Mime#20]
I'm testing the ML.Dev.Pred.Category.Sig function and encountered the following error.

> res.ici <- ML.Dev.Pred.Category.Sig(train_data = list_train_vali_Data$GSE39941,
+                                     list_train_vali_Data = list_train_vali_Data,
+                                     candidate_genes = my_genelist,
+                                     methods = c('nb','svmRadialWeights','rf','kknn','adaboost','LogitBoost','cancerclass'),
+                                     seed = 5201314,
+                                     cores_for_parallel = 60
+ )
---loading the packages ---
---loading the function---
Error in `[.data.frame`(train_data, , common_feature) : 选择了未定义的列
![1720363008774](https://github.com/l-magnificence/Mime/assets/54584775/608b7fa6-be07-4057-b5b3-44165b3c9135)

I extracted the intersection of three datasets to build a gene list ，
and using one of the datasets as the training set and the others as validation sets.
![1720363044428](https://github.com/l-magnificence/Mime/assets/54584775/d4432960-b630-43ce-9ae6-3b5ca96eaf6b)
![1720363077220](https://github.com/l-magnificence/Mime/assets/54584775/2e7c47d6-5ed6-4a92-ad9a-b2aaf5e023e7)

The R programming environment as follow

> sessionInfo()
R version 4.3.3 (2024-02-29 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 22631)

Matrix products: default

locale:
[1] LC_COLLATE=Chinese (Simplified)_China.utf8  LC_CTYPE=Chinese (Simplified)_China.utf8   
[3] LC_MONETARY=Chinese (Simplified)_China.utf8 LC_NUMERIC=C                               
[5] LC_TIME=Chinese (Simplified)_China.utf8    

time zone: Asia/Shanghai
tzcode source: internal

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] dplyr_1.1.4         cancerclass_1.48.0  binom_1.1-1.1       doParallel_1.0.17   iterators_1.0.14   
 [6] foreach_1.5.2       caret_6.0-94        lattice_0.22-6      ggplot2_3.5.1       ROCit_2.1.2        
[11] pROC_1.18.5         e1071_1.7-14        sva_3.52.0          BiocParallel_1.36.0 genefilter_1.86.0  
[16] mgcv_1.9-1          nlme_3.1-164        future_1.33.2       gridExtra_2.3       stringr_1.5.1      
[21] Mime1_0.0.0.9000    GEOquery_2.68.0     Biobase_2.62.0      BiocGenerics_0.50.0

loaded via a namespace (and not attached):
 [1] DBI_1.2.2               bitops_1.0-7            rlang_1.1.3             magrittr_2.0.3         
 [5] matrixStats_1.3.0       compiler_4.3.3          RSQLite_2.3.7           reshape2_1.4.4         
 [9] png_0.1-8               vctrs_0.6.5             pkgconfig_2.0.3         crayon_1.5.2           
[13] fastmap_1.2.0           XVector_0.42.0          utf8_1.2.4              prodlim_2023.08.28     
[17] tzdb_0.4.0              purrr_1.0.2             bit_4.0.5               zlibbioc_1.48.0        
[21] cachem_1.1.0            aplot_0.2.2             GenomeInfoDb_1.38.5     recipes_1.0.10         
[25] blob_1.2.4              R6_2.5.1                stringi_1.8.4           limma_3.58.1           
[29] rpart_4.1.23            parallelly_1.37.1       lubridate_1.9.3         Rcpp_1.0.12            
[33] future.apply_1.11.2     readr_2.1.5             IRanges_2.36.0          timechange_0.3.0       
[37] nnet_7.3-19             Matrix_1.6-5            splines_4.3.3           tidyselect_1.2.1       
[41] rstudioapi_0.16.0       timeDate_4032.109       codetools_0.2-20        listenv_0.9.1          
[45] tibble_3.2.1            plyr_1.8.9              withr_3.0.0             KEGGREST_1.40.1        
[49] gridGraphics_0.5-1      survival_3.6-4          proxy_0.4-27            xml2_1.3.6             
[53] Biostrings_2.68.1       pillar_1.9.0            MatrixGenerics_1.14.0   stats4_4.3.3           
[57] ggfun_0.1.5             generics_0.1.3          RCurl_1.98-1.14         S4Vectors_0.42.0       
[61] hms_1.1.3               munsell_0.5.1           scales_1.3.0            globals_0.16.3         
[65] xtable_1.8-4            class_7.3-22            glue_1.7.0              tools_4.3.3            
[69] data.table_1.15.4       ModelMetrics_1.2.2.2    annotate_1.78.0         gower_1.0.1            
[73] locfit_1.5-9.9          fs_1.6.4                XML_3.99-0.16.1         grid_4.3.3             
[77] tidyr_1.3.1             ipred_0.9-14            AnnotationDbi_1.62.2    edgeR_3.42.4           
[81] colorspace_2.1-0        GenomeInfoDbData_1.2.11 patchwork_1.2.0         cli_3.6.2              
[85] fansi_1.0.6             lava_1.8.0              gtable_0.3.5            yulab.utils_0.1.4      
[89] digest_0.6.35           ggplotify_0.1.2         memoise_2.0.1           lifecycle_1.0.4        
[93] hardhat_1.3.1           httr_1.4.7              statmod_1.5.0           MASS_7.3-60            
[97] bit64_4.0.5

**Comment by l-magnificence** [comment]
This is not an error of the function. Please check the format of your 3 datasets and have same colnames.

**Comment by kerwin223** [comment]
> This is not an error of the function. Please check the format of your 3 datasets and have same colnames.

Actually，I checked the format of the three datasets, and here it is:
![1720411103992](https://github.com/l-magnificence/Mime/assets/54584775/ea80f512-211d-473f-b487-1d1ba406db07)

I also checked how the genelist matches up with the column names in the three datasets.
# 获取每个数据集的列名
cols_GSE39941 <- colnames(GSE39941)
cols_GSE37250 <- colnames(GSE37250)
cols_GSE195434 <- colnames(GSE195434)

# 检查每个基因名是否在数据集中存在
exist_in_GSE39941 <- my_genelist %in% cols_GSE39941
exist_in_GSE37250 <- my_genelist %in% cols_GSE37250
exist_in_GSE195434 <- my_genelist %in% cols_GSE195434

# 创建一个数据框来显示结果
result <- data.frame(
  Gene = my_genelist,
  In_GSE39941 = exist_in_GSE39941,
  In_GSE37250 = exist_in_GSE37250,
  In_GSE195434 = exist_in_GSE195434
)
![1720411281072](https://github.com/l-magnificence/Mime/assets/54584775/f1c56984-abad-46a7-954d-5befb8d4bb10)

And this is the format of my genelist:
my_genelist <- as.character(genelist$Gene)
![1720411358759](https://github.com/l-magnificence/Mime/assets/54584775/906a67b1-3218-4911-bccd-e4eb9ac9c1a8)

**Comment by l-magnificence** [comment]
We have made some modifications, please reinstall and retry.

**Comment by kerwin223** [comment]
> We have made some modifications, please reinstall and retry.

I reinstalled package, and it works. Thanks!
